### ISSUE TITLE
Build failure: nvidia-x11-545.29.02-6.8.1

### ISSUE THREAD

**Issue by amozeo** [NixOS/nixpkgs#296773]
### Steps To Reproduce

1. build NixOS with nvidia stable drivers (545.29.02) and latest kernel (6.8.1), you can use the configuration.nix below with `NIXOS_CONFIG="path/to/configuration.nix" nixos-rebuild build`

```nix
{ config, lib, pkgs, ... }:

{
  config = {
    boot.loader.systemd-boot.enable = true;
    boot.loader.efi.canTouchEfiVariables = true;
    nixpkgs.config.allowUnfree = true;

    boot.kernelPackages = pkgs.linuxPackages_latest;
    hardware.opengl = {
      enable = true;
      driSupport = true;
      driSupport32Bit = true;
    };
    services.xserver.videoDrivers = ["nvidia"];
    hardware.nvidia = {
      modesetting.enable = true;
      powerManagement.enable = true;
      open = false;
      nvidiaSettings = true;
      package = config.boot.kernelPackages.nvidiaPackages.stable;
    };

    fileSystems."/" = {
      device = "none";
      fsType = "tmpfs";
      options = [ "defaults" "size=2G" "mode=755" ];
    };

    system.stateVersion = "23.11";
  };
}
```

### Build log

https://gist.github.com/amozeo/d3afe2946fc66c7c0bf490ca89ca7860

### Additional context

Channel: nixos-23.11

### Notify maintainers

@jonringer, @kiskae, @edwtjo - these are maintainers of `linuxKernel.packages.linux_6_8.nvidia_x11` package

### Metadata

```console
wroclaw in ~ 
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.7, NixOS, 23.11 (Tapir), 23.11.5060.617579a78725`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11"`
 - channels(wroclaw): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Kiskae** [comment]
Will probably require this patch on `release-23.11`: https://gist.github.com/joanbm/24f4d4f4ec69f0c37038a6cc9d132b43

If someone wants to test and submit a patch feel free to do so. I probably won't be able to do so until late in the evening tomorrow.

**Comment by Kiskae** [comment]
Fixed by #296968, channel tracker: https://nixpk.gs/pr-tracker.html?pr=296968
